tests/utils_test.py

```diff
@@ -89,3 +92,88 @@ def test_filter_pareto_efficient():
     )
     assert len(pareto_efficient) == 4
     assert set([0, 2, 3, 7]) == set([e.configuration["i"] for e in pareto_efficient])
+
+
+@pytest.mark.parametrize(
+    "evals",
+    (
+        [  # Different parameter values
+            Evaluation(configuration={"i": 3, "c": "A"}, objectives={"l": 0.0}),
+            Evaluation(configuration={"i": 2, "c": "B"}, objectives={"l": 0.0}),
+            Evaluation(configuration={"i": 1, "c": "C"}, objectives={"l": 0.0}),
+        ],
+        [  # Different objective values
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.0}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.3}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.2}),
+        ],
+        [  # Different parameters
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.0}),
+            Evaluation(configuration={"i": 1, "j": 2}, objectives={"l": 0.0}),
+            Evaluation(configuration={"k": 1}, objectives={"l": 0.0}),
+        ],
+        [  # Different multi objective values
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.0, "m": 0.1}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.0, "m": 0.2}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0.0, "m": 0.3}),
+        ],
+        [  # Different context
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, context={"c": 1}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, context={"c": 2}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, context={"c": 3}),
+        ],
+        [  # Different settings
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, settings={"s": 1}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, settings={"s": 2}),
+            Evaluation(configuration={"i": 1}, objectives={"l": 0}, settings={"s": 3}),
+        ],
+        [  # Different constraints
+            Evaluation(
+                configuration={"i": 1}, objectives={"l": 0}, constraints={"c": 1}
+            ),
+            Evaluation(
+                configuration={"i": 1}, objectives={"l": 0}, constraints={"c": 2}
+            ),
+            Evaluation(
+                configuration={"i": 1}, objectives={"l": 0}, constraints={"c": 3}
+            ),
+        ],
+    ),
+)
```

Comment:
If we're going for saving space with single letter names, I'd encourage you to use the patter from settings and constraints also for objectives and parameter names. Specifically, `{"o": 1.0}` or `{"o1": 1.0, "o2": -2.0}` for objectives and `{"p": 3.0}` or `{"p1": 2.0, "p2": 1.0}` for configuration. That seems to reduce the necessity to visually jump back from the value to the corresponding kwarg.

Reply:
Jup, fine for me

Reply:
Done